src/app/shared/components/product/product.component.ts

```diff
@@ -17,6 +24,8 @@ export class ProductComponent {
   @Input() product: Product;
   @Output() productImageClicked: EventEmitter<void> = new EventEmitter<void>();
 
+  constructor(public elementRef: ElementRef) {}
```

Comment:
why do we need the elementRef here?

Reply:
we need this to be able to access the elementRef in the Viewchildren for the ProductComponent.